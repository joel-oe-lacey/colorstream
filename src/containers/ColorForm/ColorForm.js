import React, { Component } from 'react';
import './ColorForm.scss';

export default class ColorForm extends Component {
    constructor() {
        super()
        this.state = { color: '' }
    }

    handleChange = e => {
        this.setState({ [e.target.name]: e.target.value })
    }

    submitColor = e => {
        e.preventDefault();
        const { selectColor } = this.props;
        selectColor(this.state.color);
        this.resetInputs();
    }

    resetInputs = () => {
        this.setState({color: ''})
    }

    render() {
        return (
            <form className='colorForm'>
                <h1>Please Enter A Color</h1>
                <input
                    type='text'
                    placeholder='color'
                    value={this.state.title}
                    name='color'
                    onChange={this.handleChange}
                />
                <button onClick={this.submitColor}>Submit</button>
            </form>
        )
    }
}