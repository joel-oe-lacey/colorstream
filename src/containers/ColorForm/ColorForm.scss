.colorForm {
    height: 45%;
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;

    h1 {
        height: 20%;
        width: 100%;
        text-align: center;
        font-size: 1.8rem;
        color: #a3b1c699;
        margin-top: 3%;
    }

    input {
        width: 40%;
        height: 30%;
        background: transparent;
        font-size: 1.2em;
        text-align: center;
        color: #a3b1c699;
        border: none;
        border-bottom: 2px solid #a3b1c699;
    }

    button {
        width: 40%;
        height: 20%;
        background-color: #E0E5EC;
        color: #a3b1c699;
        font-weight: bold;
        font-size: 1.8rem;
        border-bottom-right-radius: 5% 35%;
        border-top-right-radius: 5% 35%;
        border-bottom-left-radius: 5% 35%;
        border-top-left-radius: 5% 35%;
        box-shadow: 9px 9px 16px rgba(163, 177, 198, 0.6), -9px -9px 16px  rgba(255,255,255, 0.5);
    }
}