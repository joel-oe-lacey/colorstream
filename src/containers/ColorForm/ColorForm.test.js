import React from 'react';
import { shallow } from 'enzyme';
import ColorForm from './ColorForm';

describe('ColorForm', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<ColorForm />);
    });

    it('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
})