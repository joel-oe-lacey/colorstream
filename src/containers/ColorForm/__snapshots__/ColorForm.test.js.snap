// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorForm should match the snapshot 1`] = `
<form
  id="colorForm"
>
  <h1
    className="color-change-5x"
  >
    Color Stream
  </h1>
  <input
    name="color"
    onChange={[Function]}
    placeholder="Please Enter A Hex Color"
    type="text"
    value=""
  />
  <div
    className="outer-button"
  >
    <button
      onClick={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      Submit
    </button>
  </div>
</form>
`;
